### PR TITLE
Update Release workflow to create a git tag, github release, and publish to npm

### PR DESCRIPTION
**Pull request by pleek91**
Also requires approval for publishing new versions using the "prod" github environment

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
-      - "v[0-9]+.[0-9]+.[0-9]+" # assumes x.y.z versioning
+  push:
+    paths: 
+      - package.json
+    branches:
+      - main
 
 jobs:
+
+  check:
```
Comment (cicdw) [inline .github/workflows/release.yml:12]:
```suggestion
  version-change-check:
```
i suggest a more informative / verbose name here

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,33 +1,63 @@
...
+        id: check
+
+      - name: Cancel
+        if: steps.check.outputs.changed == 'false'
+        uses: andymckay/cancel-action@0.2
+
   build-and-publish:
+    name: Build & Publish Release
     runs-on: ubuntu-latest
+    needs: [check]
```
Comment (cicdw) [inline .github/workflows/release.yml:30]:
```suggestion
    needs: [version-change-check]
```

**Review by cicdw — COMMENTED** [review]
minor comments and suggestions, this looks nice though!

**Comment by madkinsz** [comment]
#114 seems like it should be required before this is merged, it's too much of a security hole.

**Review by cicdw — APPROVED** [review]

**Pull request merged**